src/services/validation.js

```diff
@@ -33,9 +43,25 @@ async function checkUserInFirestore(userID, validateCount, collectionName, colum
     }
 
 }
-export async function validateID(userID, validateCount, collectionName, columnName, dataSourceType){
-    if(dataSourceType == "Firestore")
+
+/** This function checks the data source to see which database API needs to be called.
+* @param {String} userID - current ID being authenticated
+* @param {Number} validateCount - indicates how many times the user has been validated
+* @param {String} collectionName - firestore collection against which the ID needs to be validated
+* @param {String} columnName - name of the column which contains the ID
+* @param {String} dataSourceType - which database needs to be used for validation
+* @param {String} authType - the authentication method the user has used
+*/
+export async function validateID(userID, validateCount, collectionName, columnName, dataSourceType, authType){
+    if(dataSourceType == "Firestore"){
+        if(authType == "ID")
     {
-        return checkUserInFirestore(userID, validateCount, collectionName, columnName)
+        return checkUserIdInFirestore(userID, validateCount, collectionName, columnName)
     }
+    else if(authType == "OTP")
+    {
+        return checkPhoneNumberInFirestore(userID, collectionName, columnName)
+    }
+}
```

Comment:
If you see, this part has incorrect indentation! I guess you need to debug why pre commit is not working